javascript/javascript1/week2/main.js

```diff
@@ -0,0 +1,184 @@
+    }
+    return `${firstname} ${surname}`;
+}
+
+getFullname("Benjamin", "Hughes"); // returns Benjamin Hughes
+getFullname(fullname1, fullname2); // returns fullname1, fullname2
+
+getFullname("Benjamin", "Hughes", true); // returns "Lord Benjamin Hughes"`
+getFullname("Benjamin", "Hughes", false); // returns "Benjamin Hughes"
+getFullname(fullname1, fullname2, true, true); // returns Lady fullname1, fullname2
+getFullname(fullname1, fullname2, false, true); // returns fullname1, fullname2
+getFullname(""); // alert for blank input
+
+//Event application
+const daysWeek = [
+    "Monday", 
+    "Tuesday", 
+    "Wednesday", 
+    "Thursday", 
+    "Friday", 
+    "Saturday", 
+    "Sunday"];
+
+function getEventWeekday(days) {
+    let currentDate = new Date(); // date from system 
+    let futureDate = currentDate.getDate() + days; // adding days to current date
+    result = futureDate % 7; // checking with modulo
+    switch (result) { // switch statement 
+        case 0:
+            console.log(daysWeek[1]);
+            break;
+        case 1:
+            console.log(daysWeek[2]);
+            break;
+        case 2:
+            console.log(daysWeek[3]);
+            break;
+        case 3:
+            console.log(daysWeek[4]);
+            break;
+        case 4:
+            console.log(daysWeek[5]);
+            break;
+        case 5:
+            console.log(daysWeek[6]);
+            break;
+        case 6:
+            console.log(daysWeek[0]);
+            break;
+        }
+        if (isNaN(result) || days === "") {// test for missing argument
+            console.log("Please, enter a date")
+        }
+        return result;
+    
+    }
+    
+    getEventWeekday(0);
+    
+    //Weather wear
+    function getDressed(temp) {
+    
+        if (temp >= 35 || temp <= -20) {
+            console.log("It is better to stay home.");
+        } else if (temp >= 20 && temp < 35) {
+            console.log("shorts and a t-shirt");
+        } else if (temp >= 10 && temp < 20) {
+            console.log("some jeans and a jaket");
+        } else if (temp > -20 && temp < 10) {
+            console.log("It is a sweater weather");
+        } else if (temp === undefined || temp === "") {
+            console.log("Please, enter a temperature");
+        }
+        return temp;
+    }
+    const clothesToWear = getDressed();
+
+//Student manager
+const class07Students = [];
+
+
+function addStudentToClass(studentName) {
+
+    if (!class07Students.includes(studentName) // excludes duplicates
+    && studentName != ``) { //  and does not add an empty string
+        class07Students.push(studentName);
+    } else if (class07Students.includes(studentName)) { // if includes duplicats does not add it
+        console.log(`Student ${studentName} is already in the class`);
+    }
+
+    if (class07Students.length <= 6 
+        && studentName === "Queen") { // always space for Queen
+            class07Students.push();
```

Comment:
what exactly are you pushing here? if you are adding 'space' for a queen, then I suppose that you need to add a student named Queen, because otherwise you would be able to add million of queen spaces